python/pyarrow/array.pxi

```diff
@@ -602,14 +604,28 @@ cdef class Array(_PandasConvertible):
         self.ap = sp_array.get()
         self.type = pyarrow_wrap_data_type(self.sp_array.get().type())
 
-    def __eq__(self, other):
-        raise NotImplementedError('Comparisons with pyarrow.Array are not '
-                                  'implemented')
-
     def _debug_print(self):
         with nogil:
             check_status(DebugPrint(deref(self.ap), 0))
 
+    def __richcmp__(self, other, int op):
+        cdef str function_name
+
+        if op == Py_EQ:
+            function_name = "equal"
+        elif op == Py_NE:
+            function_name = "not_equal"
+        elif op == Py_GT:
+            function_name = "greater"
+        elif op == Py_GE:
+            function_name = "greater_equal"
+        elif op == Py_LT:
+            function_name = "less"
+        elif op == Py_LE:
+            function_name = "less_equal"
+
+        return _pc().call_function(function_name, [self, other])
```

Comment:
Maybe factor this out into a helper function to avoid the code duplication?

Reply:
Moved the `op` to function name conversion into a helper function